python/lsst/ctrl/bps/wms/panda/panda_service.py

```diff
@@ -0,0 +1,151 @@
+        ----------
+        workflow : `~lsst.ctrl.bps.wms.htcondor.htcondor_service.HTCondorWorkflow`
+            HTCondor workflow ready to be run.
+        """
+        _LOG.debug("out_prefix = '%s'", out_prefix)
+        workflow = PanDABPSWMSWorkflow.from_generic_workflow(config, generic_workflow, out_prefix,
+                                                             f"{self.__class__.__module__}."
+                                                             f"{self.__class__.__name__}")
+        workflow.write(out_prefix)
+        return workflow
+
+    def convert_exec_string_to_hex(self, cmdline):
+        return binascii.hexlify(cmdline.encode()).decode("utf-8")
+
+    def add_decoder_prefix(self, cmdline):
+        cmdline_hex = self.convert_exec_string_to_hex(cmdline)
+        decoder_prefix = self.config.get('runner_command')
+        decoder_prefix = decoder_prefix.replace("_cmd_line_", str(cmdline_hex))
+        return decoder_prefix
+
+    def submit(self, workflow):
+        """Submit a single PanDA iDDS workflow
+
+        Parameters
+        ----------
+        workflow : `~lsst.ctrl.bps.wms_service.BaseWorkflow`
+            A single PanDA iDDS workflow to submit
+        """
+        idds_client_workflow = IDDS_client_workflow()
+        for task in workflow.generated_tasks:
+            work = DomaLSSTWork(
+                executable=self.add_decoder_prefix(task.executable),
+                primary_input_collection={'scope': 'pseudo_dataset', 'name': 'pseudo_input_collection#1'},
+                output_collections=[{'scope': 'pseudo_dataset', 'name': 'pseudo_output_collection#1'}],
+                log_collections=[], dependency_map=task.dependencies,
+                task_name=task.name,
+                task_queue=task.queue)
+            idds_client_workflow.add_work(work)
+        host = get_rest_host()
+        idds_workflow_manager = IDDS_workflow_manager(host=host)
+        request_id = idds_workflow_manager.submit(idds_client_workflow)
+        _LOG.info("Submitted into iDDs with request id=%i", request_id)
+        workflow.run_id = request_id
+
+    def report(self, wms_workflow_id=None, user=None, hist=0, pass_thru=None):
+        """Return run information based upon given constraints.
+
+        Parameters
+        ----------
+        wms_workflow_id : `int` or `str`
+            Limit to specific run based on id.
+        user : `str`
+            Limit results to runs for this user.
+        hist : `float`
+            Limit history search to this many days.
+        pass_thru : `str`
+            Constraints to pass through to HTCondor.
+
+        Returns
+        -------
+        runs : `dict` of `~lsst.ctrl.bps.wms_service.WmsRunReport`
+            Information about runs from given job information.
+        message : `str`
+            Extra message for report command to print.  This could be pointers to documentation or
+            to WMS specific commands.
+        """
+        message = ""
```

Comment:
Should probably document that this method does nothing.

Reply:
Done